src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/CreateReindexRequestHandler.cs

```diff
@@ -69,16 +73,62 @@ public async Task<CreateReindexResponse> Handle(CreateReindexRequest request, Ca
+            // Of course if we don't find a match then we leave the hash value unaffected and pass it through.
+            Dictionary<string, string> hashMap = null;
+            var searchParameterResourceTypes = new HashSet<string>();
+            if (request.TargetSearchParameterTypes.Any())
+            {
+                foreach (var searchParameterUrl in request.TargetSearchParameterTypes)
+                {
+                    SearchParameterInfo searchParameterInfo = _searchParameterDefinitionManager.GetSearchParameter(searchParameterUrl);
+                    if (searchParameterInfo == null)
+                    {
+                        throw new RequestNotValidException(Core.Resources.SearchParameterByDefinitionUriNotSupported, searchParameterUrl);
+                    }
+
+                    if (searchParameterInfo != null)
+                    {
+                        searchParameterResourceTypes.UnionWith(searchParameterInfo.BaseResourceTypes);
+                    }
+                }
+
+                hashMap = new Dictionary<string, string>();
+                foreach (var searchParamHash in _searchParameterDefinitionManager.SearchParameterHashMap)
+                {
+                    if (searchParameterResourceTypes.Contains(searchParamHash.Key))
+                    {
+                        hashMap[searchParamHash.Key] = Guid.NewGuid().ToString();
+                    }
+                    else
+                    {
+                        hashMap[searchParamHash.Key] = searchParamHash.Value;
+                    }
+                }
+            }
+
             var jobRecord = new ReindexJobRecord(
-                _searchParameterDefinitionManager.SearchParameterHashMap,
-                request.TargetResourceTypes,
-                request.MaximumConcurrency ?? _reindexJobConfiguration.DefaultMaximumThreadsPerReindexJob,
-                request.MaximumResourcesPerQuery ?? _reindexJobConfiguration.MaximumNumberOfResourcesPerQuery,
-                request.QueryDelayIntervalInMilliseconds ?? _reindexJobConfiguration.QueryDelayIntervalInMilliseconds,
-                request.TargetDataStoreUsagePercentage);
+            hashMap?.Count > 0 ? hashMap : _searchParameterDefinitionManager.SearchParameterHashMap,
+            request.TargetResourceTypes,
+            request.TargetSearchParameterTypes,
+            searchParameterResourceTypes,
+            request.MaximumConcurrency ?? _reindexJobConfiguration.DefaultMaximumThreadsPerReindexJob,
+            request.MaximumResourcesPerQuery ?? _reindexJobConfiguration.MaximumNumberOfResourcesPerQuery,
+            request.QueryDelayIntervalInMilliseconds ?? _reindexJobConfiguration.QueryDelayIntervalInMilliseconds,
+            request.TargetDataStoreUsagePercentage);
             var outcome = await _fhirOperationDataStore.CreateReindexJobAsync(jobRecord, cancellationToken);
 
             return new CreateReindexResponse(outcome);
         }
+
+        /*
```

Comment:
Should we remove these commented lines?

Reply:
Yes indeed. I missed removing that during my code cleanup.